pkg/hs/urbit-king/lib/Urbit/Vere/Eyre/Wai.hs

```diff
@@ -159,76 +166,139 @@ dupHead = do
+    ins = atomically (newLiveReq who vLiv)
+    del = atomically . rmLiveReq vLiv . fst
+
+app ::
+  HasLogFunc e =>
+  e ->
+  Ship ->
+  TVar LiveReqs ->
+  (Word64 -> ReqInfo -> STM ()) ->
+  (Word64 -> STM ()) ->
+  EyreSite ->
+  KingSubsite ->
+  W.Application
+app env who liv inform cancel site sub req respond = do
+  met <- maybe (error "bad method") pure (cookMeth req)
+  let getp = met == H.GET
+      scryFlow =
+        onException
+          (runEyreSite site req respond)
+          normalFlow
+      normalFlow =
+        runEyreSite
+          (normalSite env who liv inform cancel)
+          req
+          respond
   case W.pathInfo req of
-    ("~_~":_) -> runKingSubsite sub req respond
-    _ ->
-      runRIO env $ rwith (liveReq who liv) $ \(reqId, respApi) -> do
-        bod <- io (toStrict <$> W.strictRequestBody req)
-        met <- maybe (error "bad method") pure (cookMeth req)
-
-        let adr = reqAddr req
-            hdr = W.requestHeaders req
-            url = reqUrl req
-
-        atomically $ inform reqId $ ReqInfo adr met url hdr bod
-
-        try (sendResponse respond respApi) >>= \case
-          Right rr  -> pure rr
-          Left  exn -> do
-            atomically (cancel reqId)
-            logError $ display ("Exception during request" <> tshow exn)
-            throwIO (exn :: SomeException)
+    ("~_~" : _) -> runKingSubsite sub req respond
+    _ -> if getp then scryFlow else normalFlow
+
+normalSite ::
+  HasLogFunc e =>
+  e ->
+  Ship ->
+  TVar LiveReqs ->
+  (Word64 -> ReqInfo -> STM ()) ->
+  (Word64 -> STM ()) ->
+  EyreSite
+normalSite env who liv inform cancel = ES $ \req respond ->
+  runRIO env $
+    rwith (liveReq who liv) $ \(reqId, respApi) -> do
+      bod <- io (toStrict <$> W.strictRequestBody req)
+      met <- maybe (error "bad method") pure (cookMeth req)
+
+      let adr = reqAddr req
+          hdr = W.requestHeaders req
+          url = reqUrl req
+
+      atomically $ inform reqId $ ReqInfo adr met url hdr bod
+
+      try (sendResponse respond respApi) >>= \case
+        Right rr -> pure rr
+        Left exn -> do
+          atomically (cancel reqId)
+          logError $ display ("Exception during request" <> tshow exn)
+          throwIO (exn :: SomeException)
+
+scrySite :: HasLogFunc e => Ship -> ScryFunc -> RAcquire e EyreSite
+scrySite who scry = do
+  env <- ask
+  pure $
+    ES $ \req respond -> do
+
+      let url =  filter (/= '"') $ tshow $ reqUrl req
+          host = filter (/= '"') $ maybe "" tshow (W.requestHeaderHost req)
+      res <- runRIO env $ scryPath $ ["scry", host, url]
+      respond $ scryResp res
+  where
+    scryPath path = do
+      logDebug $ display ("scrying for " <> tshow path)
+      s <- scryOcts path
+      case s of
+        Just octs -> do
+          logDebug $ display ("scry result: " <> tshow octs)
+          pure $ unOcts octs
+        Nothing -> do 
+          logError $ display ("exception during scry on path: " <> tshow path)
```

Comment:
I'd probably log this as `logInfo` or even `logDebug`? If we're gonna try this for all requests, failure here is certainly common and expected. And we fall back, so it's not a hard error, right?